hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/ReflectionUtils.java

```diff
@@ -223,35 +222,6 @@ public synchronized static void printThreadInfo(PrintStream stream,
     
   private static long previousLogTime = 0;
     
-  /**
-   * Log the current thread stacks at INFO level.
-   * @param log the logger that logs the stack trace
-   * @param title a descriptive title for the call stacks
-   * @param minInterval the minimum time from the last 
-   */
-  public static void logThreadInfo(Log log,
```

Comment:
deprecate in 3.3

Reply:
Same as above, this can whole work behind log4j2 can only be in 4.0.0 if we follow compatibility guidelines.

Reply:
Done, updated this method as `@Deprecated` with updated javadoc. For now, this PR is good for branch-3.3 backport.

Will create another PR for removing them from trunk.